### PR TITLE
Ensure users provide valid country codes.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request makes some updates related to the lax formatting we had for `country` field.

2️⃣ Ensure we only return 2-letter country codes in the transformer. This way we won't continue to send bad data to Drupal when users with unsanitized values try to log in.

🔀 Fixes the user profile form to redirect you to the homepage after changing your profile, rather than to the unimplemented `show` route (a blank page)!

🗺 Adds the League's [`iso3166 `](https://github.com/thephpleague/iso3166) package and uses it to display [a dropdown "country" field](https://cloud.githubusercontent.com/assets/583202/25401086/b395401a-29c2-11e7-9f0e-bb5bd7aabd65.png) for users rather than an open-ended text field.

👁 Adds a mutator (to standardize uppercase values) and a `country` validator so that API clients can't accidentally set bad data on the `country` field.

References DoSomething/phoenix#7304.

#### How should this be reviewed?
Should be good to peek commit-by-commit. I also added a lil' functional test at the end.

Here's the [Wercker build](https://app.wercker.com/dosomething/northstar/runs/build/58ff97b6cc05940001b2b6a5) since it no longer links automatically. 😕

#### Checklist
- [ ] Documentation added for changed endpoints.
- [x] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by weerd** [comment]
> the League™'s

File: app/Http/Transformers/UserTransformer.php [hunk 1]
```diff
@@ -55,7 +55,9 @@ public function transform(User $user)
         }
 
         $response['language'] = $user->language;
-        $response['country'] = $user->country;
+
+        // Ensure we only return 2-letter country codes.
+        $response['country'] = strlen($user->country) <= 2 ? $user->country : null;
```
Comment (weerd) [inline app/Http/Transformers/UserTransformer.php]: Can the country code be 1 char?

File: app/Http/Controllers/Web/UserController.php [hunk 2]
```diff
@@ -51,9 +50,8 @@ public function home()
      */
     public function show($id)
     {
-        $user = User::findOrFail($id);
-
-        $authUser = auth()->guard('web')->user()->toArray();
+        // @TODO: Implement this route.
```
Comment (weerd) [inline app/Http/Controllers/Web/UserController.php]: 😬

File: app/helpers.php [hunk 3]
```diff
@@ -93,3 +93,13 @@ function client_id()
     // If not an API request, use Client ID from `/authorize` call or just 'northstar'.
     return session('authorize_client_id', 'northstar');
 }
+
+/**
+ * Get a list of countries keyed by ISO country code.
+ */
+function get_countries()
+{
+    $iso = (new League\ISO3166\ISO3166)->getAll();
```
Comment (weerd) [inline app/helpers.php]: coooool!

File: app/Http/Transformers/UserTransformer.php (hunk 1 above)
Comment (DFurnes) [inline app/Http/Transformers/UserTransformer.php]: Ideally not! Yeah, lemme change this to a `==`.

Comment (DFurnes) [inline app/Http/Transformers/UserTransformer.php]: Updated in ef4b379!

**Pull request merged**